enterprise/metrics/src/test/java/org/neo4j/metrics/source/ClusterMetricsTest.java

```diff
@@ -0,0 +1,128 @@
+package org.neo4j.metrics.source;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.ScheduledReporter;
+import com.codahale.metrics.Timer;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.SortedMap;
+import java.util.concurrent.TimeUnit;
+
+import org.neo4j.cluster.InstanceId;
+import org.neo4j.helpers.Settings;
+import org.neo4j.helpers.collection.MapUtil;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.ha.cluster.HighAvailabilityMemberState;
+import org.neo4j.kernel.ha.cluster.HighAvailabilityMemberStateMachine;
+import org.neo4j.kernel.ha.cluster.HighAvailabilityModeSwitcher;
+import org.neo4j.kernel.ha.cluster.member.ClusterMember;
+import org.neo4j.kernel.ha.cluster.member.ClusterMembers;
+import org.neo4j.kernel.ha.cluster.member.ObservedClusterMembers;
+import org.neo4j.kernel.lifecycle.LifeSupport;
+import org.neo4j.metrics.MetricsSettings;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+import static org.neo4j.helpers.collection.MapUtil.stringMap;
+
+
+public class ClusterMetricsTest
+{
+    @Test
+    public void clusterMetricsReportMasterAvailable() throws Exception
+    {
+        // given
+        MetricRegistry metricRegistry = new MetricRegistry();
+        Config config = new Config( stringMap( MetricsSettings.clusterEnabled.name(), Settings.TRUE ) );
+        ClusterMembers clusterMembers =
+                getClusterMembers( HighAvailabilityModeSwitcher.MASTER, HighAvailabilityMemberState.MASTER );
+
+        LifeSupport life = new LifeSupport();
+        life.add( new ClusterMetrics( metricRegistry, config, clusterMembers ) );
+        life.start();
+
+        // when
+        TestReporter reporter = new TestReporter( metricRegistry );
+        reporter.start( 10, TimeUnit.MILLISECONDS );
+
+        // then wait for the reporter to get a report
+        reporter.awaitReport();
```

Comment:
Call the `report()` method of the `ScheduledReporter` instead of starting a background thread (that you don't stop) and waiting for it to do stuff.